app/src/main/java/org/mozilla/fenix/components/appstate/AppStoreReducer.kt

```diff
@@ -148,9 +148,15 @@ internal object AppStoreReducer {
             pocketStories = emptyList(),
             pocketSponsoredStories = emptyList()
         )
-        is AppAction.PocketSponsoredStoriesChange -> state.copy(
-            pocketSponsoredStories = action.sponsoredStories
-        )
+        is AppAction.PocketSponsoredStoriesChange -> {
```

Comment:
I'm not familiar with how the Pocket code works. Does updating the setting necessitate updating the underlying data while on settings? I would have normally expected it to just be part of the Homescreen rendering process.

Reply:
The current flow assumes the stories composable just [observes the `pocketStories` state](https://github.com/mozilla-mobile/fenix/blob/64a12574b4a0ebd64cca91c0c9755f4928dc9972/app/src/main/java/org/mozilla/fenix/home/pocket/PocketStoriesViewHolder.kt#L56).
That is updated through `pocketStories = updatedStoriesState.getFilteredStories()` calls being done in the reducer.
If we emit `PocketSponsoredStoriesChange(emptyList())` from settings when the user disables the feature the reducer will update the AppState such that when the user goes back to homescreen and the composables read `pocketStories` it will only contain recommended stories.